### PR TITLE
mgmt, appservice, support async deploy with deployment ID

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Fix https://github.com/Azure/azure-sdk-for-java/issues/24934

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@Flanker32 

Adding Kudu part of the deployment status related code here.

However appservice swagger does not have the deployment status API and they've mentioned they might modify the model.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java [hunk 1]
```diff
@@ -49,4 +65,78 @@ public void canDeployZip() throws Exception {
...
+            connection.disconnect();
+        }
+        KuduDeploymentResult deployResult =
+            webApp1.pushDeploy(DeployType.JAR, jarFile, new DeployOptions().withTrackDeployment(true));
+
+        Assertions.assertNotNull(deployResult.deploymentId());
+
+        // poll deployment status
+        String deploymentStatusUrl = AzureEnvironment.AZURE.getResourceManagerEndpoint()
+            + "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/deploymentStatus/{deploymentId}?api-version=2021-03-01";
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java:104]: I noticed they have a swagger for getting deployment by deploymentId, which includes deployment status as a part of the response: 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json#L2880-L2927

Will it help simplify?

**Review by XiaofeiCao — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java:104]: It appears not a same resource.

This one looks like below
```
{ 
  "id": "freeberg-java-deploy-test", 
  "name": "Microsoft.Web/sites", 
  "type": "Central US EUAP", 
  "properties": { 
    "id": "6e659b28-d0b0-4fd1-b216-e5655f238777", 
    "buildStatus": "RuntimeStarting", 
    "numberOfInstancesInProgress": 0, 
    "numberOfInstancesSuccessful": 6, 
    "numberOfInstancesFailed": 2, 
    "failedInstancesLogs": null 
  } 
} 
```

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java [hunk 2]
```diff
@@ -3,38 +3,54 @@
...
 
 public class OneDeployTests extends AppServiceTest {
 
-    @Test
-    @DoNotRecord
-    public void canDeployZip() throws Exception {
-        if (skipInPlayback()) {
-            return;
-        }
+    private static final String GATEWAY_JAR_URL = "https://github.com/weidongxu-microsoft/azure-sdk-for-java-management-tests/raw/master/spring-cloud/gateway.jar";
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java:40]: Notice this is a path under personal repo, can others run the LIVE test in the future using this GATEWAY_JAR_URL ?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/OneDeployTests.java:40]: Yes they can. Just throw this weidongxu-microsoft account to MS.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**